### PR TITLE
Updating the PM interface to defined names for variables & constraints

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

File: src/network_models/powermodels_interface.jl [hunk 1]
```diff
@@ -457,7 +459,7 @@ function add_pm_con_refs!(
         if pm_v in pm_con_names
             container = PSI.add_cons_container!(
                 psi_container,
-                ps_v,
+                PSI.constraint_name(ps_v, PSY.Bus),
```
Comment (daniel-thom) [inline src/network_models/powermodels_interface.jl:462]:
```suggestion
                make_constraint_name(ps_v, PSY.Bus),
```

File: test/test_network_constructors.jl [hunk 2]
```diff
@@ -118,7 +118,11 @@ end
     c_sys14_dc = build_system("c_sys14_dc")
     systems = [c_sys5, c_sys14, c_sys14_dc]
     objfuncs = [GAEVF, GQEVF, GQEVF]
-    constraint_names = [:RateLimit_ub__Line, :RateLimit_lb__Line, :Bus_active_power_balance]
+    constraint_names = [
+        :RateLimit_ub__Line,
+        :RateLimit_lb__Line,
+        PSI.constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus),
```
Comment (daniel-thom) [inline test/test_network_constructors.jl:124]:
```suggestion
        make_constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus),
```

File: test/test_network_constructors.jl [hunk 3]
```diff
@@ -169,8 +173,8 @@ end
     constraint_names = [
         :RateLimitFT__Line,
         :RateLimitTF__Line,
-        :Bus_active_power_balance,
-        :Bus_reactive_power_balance,
+        PSI.constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus),
+        PSI.constraint_name(PSI.NODAL_BALANCE_REACTIVE, PSY.Bus),
```
Comment (daniel-thom) [inline test/test_network_constructors.jl:177]:
```suggestion
        make_constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus),
        make_constraint_name(PSI.NODAL_BALANCE_REACTIVE, PSY.Bus),
```

**Comment by sourabhdalvi** [comment]
> We are in the process of changing the function name. Please use the new `make_constraint_name`.

After talking to Dan, the new `make_constraint_name` isn't currently on this branch, so I'm sticking to `constraint_name` and Dan will update them once this PR is merged

File: test/test_network_constructors.jl (hunk 2 above)
Comment (jd-lara) [inline test/test_network_constructors.jl:124]: This test seems that it needs refinement. Why is this the only constraint name tested this way?

File: test/test_network_constructors.jl [hunk 4]
```diff
@@ -169,8 +173,8 @@ end
     constraint_names = [
         :RateLimitFT__Line,
         :RateLimitTF__Line,
-        :Bus_active_power_balance,
-        :Bus_reactive_power_balance,
+        PSI.constraint_name(PSI.NODAL_BALANCE_ACTIVE, PSY.Bus),
```
Comment (jd-lara) [inline test/test_network_constructors.jl:176]: same here

File: test/test_network_constructors.jl (hunk 2 above)
Comment (sourabhdalvi) [reply test/test_network_constructors.jl:124]: I'm not sure, I didn't write these these probably Clayton did.

**Comment by jd-lara** [comment]
@daniel-thom we can merge this and add it to the general fixes on constraint names later on.

**Pull request merged**